File: backend/dataset/maxPathSum/maxPathSum_19.py
class Solution:
    def maxPathSum(self, root: TreeNode) -> int:
        Max = float("-inf")
        def maxPath(root):
            nonlocal Max
            if not root:
                return float("-inf")
            left1= maxPath(root.left)
            right1= maxPath(root.right)
            temp = max(root.val, root.val + left1, root.val + right1)
            Max = max(Max, temp, left1+right1+root.val)
            return temp
        res = maxPath(root)
        return Max if Max != float("-inf") else 0

File: backend/CDataset/maxPathSum/maxPathSum_19.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
static int speedup=[](){
	ios_base::sync_with_stdio(false);
	cin.tie(nullptr);
	return 0;
}();

class Solution {
public:
    class TreeNode{
        public:
        TreeNode* left;
        TreeNode* right;
        int val;
    };
	int maxi=INT_MIN;

	int maxSum(TreeNode* root) {
		if(!root) return 0;
		if(!root->left && !root->right) {
			maxi = max(maxi,root->val);
			return root->val;
		} 

		int v1 = maxSum(root->left);
		int v2 = maxSum(root->right);
		int tmp = max(root->val, max(v1, v2)+root->val);
		maxi = max(maxi, max( (v1+v2+root->val), tmp));
		return tmp;
	}

	int maxPathSum(TreeNode* root) {
		maxSum(root);
		return maxi;
	}
};